filestorage.go

```diff
@@ -284,25 +304,36 @@ type fileStorageWaiter struct {
 	wg       *sync.WaitGroup
 }
 
-// Wait waits until the lock is released.
+// Wait waits until the lock at fw.filename is
+// released or until it becomes stale.
 func (fw *fileStorageWaiter) Wait() {
 	start := time.Now()
 	fw.wg.Wait()
-	for time.Since(start) < 1*time.Hour {
-		_, err := os.Stat(fw.filename)
-		if os.IsNotExist(err) {
+	for time.Since(start) < staleLockDuration {
+		info, err := os.Stat(fw.filename)
+		if err != nil {
+			return
+		}
+		if fileLockIsStale(info) {
 			return
 		}
 		time.Sleep(1 * time.Second)
 	}
 }
 
-var fileStorageNameLocks = make(map[string]*fileStorageWaiter)
-var fileStorageNameLocksMu sync.Mutex
+func fileLockIsStale(info os.FileInfo) bool {
+	if info == nil {
+		return true
+	}
+	return time.Since(info.ModTime()) > staleLockDuration
+}
 
 var _ Storage = FileStorage{}
-var _ Waiter = &fileStorageWaiter{}
 
 // staleLockDuration is the length of time
 // before considering a lock to be stale.
 const staleLockDuration = 2 * time.Hour
+
+// fileStorageNameLocksMu guard access to all
+// FileStorage.fileStorageNameLock values.
+var fileStorageNameLocksMu sync.Mutex
```

Comment:
This being global means it will block *all* instances, which is probably not a desirable behavior.

Reply:
Yes, but the locks are brief, and this was done to avoid copying mutex values also.

Reply:
There shouldn't be any possibility of copying the mutex (and `get vet` should catch that).